src/test/java/study/StringAddCalculator.java

```diff
@@ -0,0 +1,87 @@
+package study;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+    public static int splitAndSum(String text) {
+        if (emptyOrNull(text)){
+            return 0;
+        }
+        if (singleNum(text)) {
+            return Integer.parseInt(text);
+        }
+
+        Matcher pattern = Pattern.compile("//(.)\n(.*)").matcher(text);
+
+        int patternSum = 0;
+        int notPatternSum =0 ;
+
+        if (isPatternMatched(pattern)) {
+            patternSum = sumPattern(pattern);
+        } else {
+            notPatternSum = sumAll(text);
+        }
+        if (patternSum >0){
+            return  patternSum;
+        }
+        if (notPatternSum>0){
+            return notPatternSum;
+        }
+
+        return -1;
+    }
+
+    private static boolean emptyOrNull(String text) {
+        boolean isEmptyOrNull = false;
+        if (text == null || text.isEmpty()) {
+            isEmptyOrNull = true;
+        }
+        return isEmptyOrNull;
+    }
+
+    private static boolean singleNum(String text){
+        boolean isSingleNum = false;
+        if (text.length() == 1){
+            isSingleNum = true;
+        }
+        return isSingleNum;
```

Comment:
```suggestion
        return text.length() == 1;
```
이 메서드처럼 다른 메서드도 간단하게 구현이 가능할 것 같네요ㅎㅎ